docs/quickstart.md

```diff
@@ -45,29 +32,74 @@ az account list
+Before running any of the scripts, the bootstrap configuration variables need to be set. This is done in an `.env` file, and this file is read and parsed by scripts.
 
-To initialise terraform, from the root of the repository run:
+Note. `.tfvars` file is not used, this is intentional. The dotenv format is easier to parse, meaning we can use the values for bash scripts and other purposes
 
-```cmd
-cd templates/core/terraform
-terraform init
-```
+To change to the directory containing the bootstrap scripts for a terraform envionment:
 
-Create execution plan using your variables file created earlier. Specify a plan file name, we will use `azuretre-dev.tfplan`:
+- In bash run `cd ./bootstrap/terraform`
 
-```cmd
-terraform plan -var-file dev.tfvars -out azuretre-dev.tfplan
-```
+Copy [.env.sample](../bootstrap/terraform/.env.sample) to `.env` in the  and set values for all variables:
 
-You will be prompted for your GitHub username and PAT token.
+- `TF_VAR_state_storage` - The name of the storage account to hold Terraform state.
+- `TF_VAR_mgmt_res_group` - The shared resource group for all hub resources, including the storage account.
+- `TF_VAR_state_container` - Name of the blob container to hold Terraform state (default: `tfstate`).
+- `TF_VAR_prefix` - A prefix added to all resources, pick your project name or other prefix to give the resources unique names.
+- `TF_VAR_location` - Azure region to deploy all resources into.
+- `TF_VAR_image_tag` - Default tag for docker images that will be pushed to the container registry and deployed with the TRE
 
-Apply the plan:
-```cmd
-terraform apply azuretre-dev.tfplan
-```
+### Bootstrap of Backend State
+
+As a principal we want all our resources defined in Terraform, including the storage account using by Terraform to hold backend state. This results in a chicken and egg problem.
+
+To solve this a bootstrap script is used which creates the initial storage account and resource group using the Azure CLI. Then Terraform is initialized using this storage account as a backend, and the storage account imported into state
+
+- From bash run `./bootstrap.sh`
+
+This script should never need running a second time even if the other management resources are modified
+
+### Management Resource Deployment
+
+The deployment of the rest of the shared management resources is done via Terraform, and the various .tf files in the root of this repo.
+
+- From bash run `./deploy.sh`
+
+This Terraform creates & configures the following:
+
+- Resource Group (also in bootstrap).
+- Storage Account for holding Terraform state (also in bootstrap).
+- Azure Container Registry.
+
+### Build and push docker images
+
+To build and push the docker images run:
+
+- From bash run `./build_and_publish_docker_images.sh.sh`
+
+This Terraform creates & configures the following:
+
+- Resource Group (also in bootstrap).
+- Storage Account for holding Terraform state (also in bootstrap).
+- Azure Container Registry.
```

Comment:
Looking at the code, this documentation doesn't match what the script does (build_and_publish.. doesn't create rg, etc.)?

Reply:
Sorry, more haste...

Reply:
Fixed